### DISCUSSION TITLE
build error

### DISCUSSION THREAD

**Discussion by PerrywLiu** [filecoin-project/lotus#7669] [Build and Sync]
I build lotus 1.13.0 on MacOS 11.0.1 (20B50)，become error.
but when I build 1.11.2rc can success.
I have update rust to the latest version.
what should I do?
the error as follow:

`[INFO  cargo_lipo::lipo] Building "filcrypto" for "aarch64-apple-darwin"
   Compiling cfg-if v1.0.0
   Compiling lazy_static v1.4.0
   Compiling scopeguard v1.1.0
   Compiling ppv-lite86 v0.2.10
   Compiling byteorder v1.4.3
   Compiling tap v1.0.1
   Compiling funty v1.2.0
   Compiling either v1.6.1
'+adx' is not a recognized feature for this target (ignoring feature)
'+avx2' is not a recognized feature for this target (ignoring feature)
'+adx' is not a recognized feature for this target (ignoring feature)
'+adx' is not a recognized feature for this target (ignoring feature)
'+avx2' is not a recognized feature for this target (ignoring feature)
'+avx2' is not a recognized feature for this target (ignoring feature)
'+adx' is not a recognized feature for this target (ignoring feature)
'+avx2' is not a recognized feature for this target (ignoring feature)
'+adx' is not a recognized feature for this target (ignoring feature)
'+avx2' is not a recognized feature for this target (ignoring feature)
'+adx' is not a recognized feature for this target (ignoring feature)
'+adx' is not a recognized feature for this target (ignoring feature)
'+avx2' is not a recognized feature for this target (ignoring feature)
'+avx2' is not a recognized feature for this target (ignoring feature)
error[E0463]: can't find crate for `core`
  |
  = note: the `aarch64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
'+adx' is not a recognized feature for this target (ignoring feature)
error[E0463]: can't find crate for `core`
  |
  = note: the `aarch64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
'+avx2' is not a recognized feature for this target (ignoring feature)
error[E0463]: can't find crate for `core`
  |
  = note: the `aarch64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error[E0463]: can't find crate for `core`
  |
  = note: the `aarch64-apple-darwin` target may not be installed

error[E0463]: can't find crate for `core`
  |
  = note: the `aarch64-apple-darwin` target may not be installed

error: aborting due to previous error

error: could not compile `lazy_static`

To learn more, run the command again with --verbose.
warning: build failed, waiting for other jobs to finish...
For more information about this error, try `rustc --explain E0463`.
error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error[E0463]: can't find crate for `core`
  |
  = note: the `aarch64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error[E0463]: can't find crate for `std`
  |
  = note: the `aarch64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error[E0463]: can't find crate for `std`
  |
  = note: the `aarch64-apple-darwin` target may not be installed

error: aborting due to previous error

For more information about this error, try `rustc --explain E0463`.
error: build failed
[ERROR cargo_lipo] Failed to build "filcrypto" for "aarch64-apple-darwin": Executing "/Users/liupw/.rustup/toolchains/nightly-2021-04-24-x86_64-apple-darwin/bin/cargo" "--color" "auto" "build" "-p" "filcrypto" "--target" "aarch64-apple-darwin" "--release" "--lib" "--no-default-features" "--features" "multicore-sdr,opencl" finished with error status: exit status: 101
+ rm -f /var/folders/qy/p4sw2gnd33d3pv4kr2d4y1m80000gn/T/tmp.qMZZg4FL
make[1]: *** [.install-filcrypto] Error 1
make: *** [build/.filecoin-install] Error 2`

**Comment by SgtCoin** [comment]
@PerrywLiu Can you run `rustup target add aarch64-apple-darwin`?

**Reply by alvin-reyes** [reply]
Got the same error even after running `rustup target add aarch64-apple-darwin`
```
[ERROR cargo_lipo] Failed to build "filcrypto" for "aarch64-apple-darwin": Executing "/Users/brambear/.rustup/toolchains/nightly-2021-07-29-x86_64-apple-darwin/bin/cargo" "--color" "auto" "build" "-p" "filcrypto" "--target" "aarch64-apple-darwin" "--release" "--lib" "--no-default-features" "--features" "multicore-sdr,opencl" finished with error status: exit status: 101
```

**Comment by SgtCoin** [comment]
@PerrywLiu 

Did you make any progress with the build?

**Comment by JairoTan** [comment]
hey bro，Has the problem been solved？

**Comment by SgtCoin** [comment]
@PerrywLiu 

Did you upgrade to 1.14.4 and retry?

**Reply by JairoTan** [reply]
This is my version

**Reply by SgtCoin** [reply]
Can you try the last few steps in [https://github.com/filecoin-project/filecoin-ffi/issues/194](https://github.com/filecoin-project/filecoin-ffi/issues/194)

**Comment by zhuaiballl** [comment]
I have rustup 1.24.3. The error still exists after I running `rustup target add aarch64-apple-darwin`.
